projects/valtimo/components/src/lib/components/form-io/form-io-address/form-io-address.component.ts

```diff
@@ -0,0 +1,191 @@
+          <div class="input-group">
+            <input cdsText id="postalCode" type="text" formControlName="postalCode" mask="0000 SS" (input)="toUpper($event)">
+            <div class="validation-message" *ngIf="this.required && addressForm.controls.postalCode.invalid && this.addressForm.get('postalCode').value === '' && addressForm.controls.postalCode.touched">{{ 'custom-form-io-components.form-io-address.postal-code' | translate }} {{ 'custom-form-io-components.isRequired' | translate }}</div>
+            <div class="validation-message" *ngIf="this.required && addressForm.controls.postalCode.invalid && this.addressForm.get('postalCode').value !== '' && addressForm.controls.postalCode.touched">{{ 'custom-form-io-components.form-io-address.postal-code' | translate }} {{ 'custom-form-io-components.isInvalid' | translate }}</div>
+          </div>
+        </cds-label>
+
+        <cds-label for="city" class="input-large">
+          {{ 'custom-form-io-components.form-io-address.city' | translate }}:
+          <div class="input-group">
+            <input cdsText id="city" type="text" formControlName="city">
+            <div class="validation-message" *ngIf="this.required && addressForm.controls.city.invalid && addressForm.controls.city.touched">{{ 'custom-form-io-components.form-io-address.city' | translate }} {{ 'custom-form-io-components.isRequired' | translate }}</div>
+          </div>
+        </cds-label>
+      </div>
+    </div>
+
+  `,
+  styles: [`
+    .address-container {
+      display: flex;
+      flex-direction: column;
+      gap: 10px;
+    }
+    .address-row {
+      display: flex;
+      gap: 10px;
+    }
+    .street {
+      flex: 3;
+    }
+    .input-small {
+      flex: 1;
+    }
+    .input-large {
+      flex: 2;
+    }
+    .input-group {
+      display: flex;
+      flex-direction: column;
+    }
+    .validation-message {
+        color: #dc3545;
+        font-size: 14.4px;
+        margin-top: 2px;
+    }
+  `]
+})
+export class FormIoAddressComponent implements FormioCustomComponent<any> {
+  @Input() value: any;
+  @Input() disabled: boolean = false;
+  @Input() mask: boolean = false;
+  @Input() required: boolean = false;
+  @Input() requiredStreet: boolean = false;
+  @Input() requiredHouseNumber: boolean = false;
+  @Input() requiredHouseLetter: boolean = false;
+  @Input() requiredHouseNumberAddition: boolean = false;
+  @Input() requiredPostalCode: boolean = false;
+  @Input() requiredCity: boolean = false;
+  @Input() keyLanguage: any;
+  @Output() valueChange = new EventEmitter<any>();
+
+  addressForm = new FormGroup({
+    street: new FormControl(''),
+    houseNumber: new FormControl(''),
+    houseLetter: new FormControl(''),
+    houseNumberAddition: new FormControl(''),
+    postalCode: new FormControl(''),
+    city: new FormControl('')
+  });
+
+  constructor() {
+    console.log('this: ', this);
+    setTimeout( () => {
```

Comment:
move all code (together with settimeout block) to a separate function i.e. private prefillForm(). Call this function in ngOnInit or ngAfterViewInit; https://stackoverflow.com/questions/35763730/difference-between-constructor-and-ngoninit (avoid constructor when possible)